### ISSUE TITLE
Feature: Support for Multipart Requests

### ISSUE THREAD

**Issue by TheHemantKaushik** [CodingAleCR/http_interceptor#6]
Getting error when trying to send multipart request like this. No header is added. 

```dart
    var request = MultipartRequest(
      "POST",
      Uri.parse("url"),
    );
    var file = await MultipartFile.fromPath("name", "path");
    request.files.add(file);
    var response = await client.send(request);
```

How can I send multipart request with all headers added in interceptor?

**Comment by CodingAleCR** [comment]
Hey, I might need to look into this, in theory they should be applied all but to be honest I did not think to implement this at the start, I'll try to look into it and get back to you as soon as I can to figure out if it is something in the roadmap or if it is something we can do currently.

**Comment by CodingAleCR** [comment]
As of right now this plugin does not support MultipartRequests. I will add it to the roadmap since it is a huge improvement. Cheers!

**Comment by januwA** [comment]
Hi, try using [this branch](https://github.com/januwA/http_interceptor), I added the `postFile` method to `client` to simplify sending files. 😃

```
dependencies:
  http_interceptor:
    git:
      url: https://github.com/januwA/http_interceptor.git
      ref: master
```

**Comment by chitgoks** [comment]
@januwA  great job! just what i was looking for! Got stuck with MultiPartRequest then tried to open up a new issue and found this is already existing.

**Comment by CodingAleCR** [comment]
Agreed, I'm sorry for not being able to look into this and add it. I'm aiming for a 0.3.0 release with this in. Hopefully it happens soon.

**Comment by CodingAleCR** [comment]
This has been added to the private backlog that I will use for handling tasks, features and bugs. I will close this and refer back to it once the feature is ready.
